lib/vmpooler/pool_manager.rb

```diff
@@ -683,36 +752,59 @@ def _check_pool(pool, provider)
         end
       end
 
+      # Create template delta disks
+      prepare_template(pool, provider)
```

Comment:
if we just want to do this on startup, maybe we could move it out of this running loop and go over each pool on startup? that would remove multiple thread access.

Reply:
We could, though if it can happen in parallel I think startup time should be better. I had some brief internal debate about whether it worked better in check_pool, or not, and arrived at this being the solution that worked. I'm open to consider it further though if this doesn't make sense.